File: 962. Maximum Width Ramp.py

class Solution:
    def maxWidthRamp(self, nums: List[int]) -> int:
        n = len(nums)
        indices = [i for i in range(n)]

        indices.sort(key=lambda i: (nums[i], i))
        print(indices)

        min_index = n
        max_width = 0

        for i in indices:
            max_width = max(max_width, i-min_index)
            min_index = min(min_index, i)

        return max_width

    """
    def maxWidthRamp(self, nums: List[int]) -> int:
        max_width = 0

        for i in range(len(nums)-1):
            if max_width >= len(nums)-1-i:
                return max_width

            for j in range(max_width-i+1, len(nums)):
                width = j-i
                if nums[j] >= nums[i]:
                    max_width = max(max_width, width)

        return max_width
    """

File: 962. Maximum Width Ramp.rs

impl Solution {
    pub fn max_width_ramp(nums: Vec<i32>) -> i32 {
        let mut max_width : i32 = 0;
        let mut sorted_nums = nums.clone();
        sorted_nums.sort();

        max_width
    }
}

/*
impl Solution {
    pub fn max_width_ramp(nums: Vec<i32>) -> i32 {
        let mut max_width : i32 = 0;

        for i in (0..nums.len()-1) {
            for j in (i+1..nums.len()) {
                if nums[i] <= nums[j] {
                    let width : i32 = (j-i) as i32;
                    if max_width < width {
                        max_width = width;
                    }
                }
            }
        }

        max_width
    }
}
*/